app/controllers/products_common_controller.rb

```diff
@@ -41,60 +41,19 @@ def index
   # GET /facilities/:facility_id/(services|items|bundles)/:(service|item|bundle)_id
   # TODO InstrumentsController#show has a lot in common; refactor/extract/consolidate
   def show
-    assert_product_is_accessible!
-    add_to_cart = true
-    @login_required = false
-
-    # does the product have active price policies?
-    unless @product.available_for_purchase?
-      add_to_cart = false
-      @error = "not_available"
-    end
-
-    # is user logged in?
-    if add_to_cart && acting_user.blank?
-      @login_required = true
-      add_to_cart = false
-    end
-
-    # when ordering on behalf of, does the staff have permissions for this facility?
-    if add_to_cart && acting_as? && !session_user.operator_of?(@product.facility)
-      add_to_cart = false
-      @error = "not_authorized_acting_as"
-    end
-
-    # does the user have a valid payment source for purchasing this reservation?
-    if add_to_cart && acting_user.accounts_for_product(@product).blank?
-      add_to_cart = false
-      @error = "no_accounts"
-    end
-
-    # does the product have any price policies for any of the groups the user is a member of?
-    if add_to_cart && !price_policy_available_for_product?
-      add_to_cart = false
-      @error = "not_in_price_group"
-    end
-
-    # is the user approved?
-    if add_to_cart && !@product.can_be_used_by?(acting_user) && !session_user_can_override_restrictions?(@product)
-      if SettingsHelper.feature_on?(:training_requests)
-        if TrainingRequest.submitted?(session_user, @product)
-          flash[:notice] = t_model_error(Product, "already_requested_access", product: @product)
-          return redirect_to facility_path(current_facility)
-        else
-          return redirect_to new_facility_product_training_request_path(current_facility, @product)
-        end
-      else
-        add_to_cart = false
-        @error = "requires_approval"
-      end
+    @show_product = ShowProduct.new(self, @product, acting_user, session_user, acting_as?)
+    @add_to_cart = @show_product.able_to_add_to_cart?
```

Comment:
I'm guessing setting these as instance variables instead of just using them directly is to maintain compatibility with the specs?